cyral/resource_cyral_integration_aws_iam_authn.go

```diff
@@ -0,0 +1,179 @@
+	stringARNs := make([]string, 0, len(arns))
+	for _, arn := range arns {
+		stringARNs = append(stringARNs, arn.(string))
+	}
+
+	wrapper.Integration.IAMRoleARNs = stringARNs
+	return nil
+}
+
+type CreateAWSIAMIntegrationResponse struct {
+	ID string `json:"id"`
+}
+
+func (c *CreateAWSIAMIntegrationResponse) WriteToSchema(d *schema.ResourceData) error {
+	d.SetId(c.ID)
+	return nil
+}
+
+var ReadAWSIAMIntegration = ResourceOperationConfig{
+	Name:       "AWSIAMIntegrationRead",
+	HttpMethod: http.MethodGet,
+	CreateURL: func(d *schema.ResourceData, c *client.Client) string {
+		return fmt.Sprintf(
+			"https://%s/v1/integrations/aws/iam/%s",
+			c.ControlPlane,
+			d.Id(),
+		)
+	},
+	NewResponseData: func(_ *schema.ResourceData) ResponseData {
+		return &APIWrapper{}
+	},
+	RequestErrorHandler: &ReadIgnoreHttpNotFound{resName: "AWS IAM AuthN Integration"},
+}
+
+func resourceIntegrationAWSIAMAuthN() *schema.Resource {
+	return &schema.Resource{
+		Description: "Authenticate users based on AWS IAM credentials",
+		CreateContext: CreateResource(
+			ResourceOperationConfig{
+				Name:       "AWSIAMIntegrationCreate",
+				HttpMethod: http.MethodPost,
+				CreateURL: func(d *schema.ResourceData, c *client.Client) string {
+					return fmt.Sprintf("https://%s/v1/integrations/aws/iam", c.ControlPlane)
+				},
+				NewResourceData: func() ResourceData {
+					return &APIWrapper{}
+				},
+				NewResponseData: func(_ *schema.ResourceData) ResponseData {
+					return &CreateAWSIAMIntegrationResponse{}
+				},
+			},
+			ReadAWSIAMIntegration,
+		),
+		ReadContext: ReadResource(ReadAWSIAMIntegration),
+		UpdateContext: UpdateResource(
+			ResourceOperationConfig{
+				Name:       "AWSIAMIntegrationUpdate",
+				HttpMethod: http.MethodPut,
+				CreateURL: func(d *schema.ResourceData, c *client.Client) string {
+					return fmt.Sprintf(
+						"https://%s/v1/integrations/aws/iam/%s",
+						c.ControlPlane,
+						d.Id(),
+					)
+				},
+				NewResourceData: func() ResourceData {
+					return &APIWrapper{}
+				},
+			},
+			ReadAWSIAMIntegration,
+		),
+		DeleteContext: DeleteResource(
+			ResourceOperationConfig{
+				Name:       "AWSIAMIntegrationDelete",
+				HttpMethod: http.MethodDelete,
+				CreateURL: func(d *schema.ResourceData, c *client.Client) string {
+					return fmt.Sprintf(
+						"https://%s/v1/integrations/aws/iam/%s",
+						c.ControlPlane,
+						d.Id(),
+					)
+				},
+			},
+		),
+
+		Importer: &schema.ResourceImporter{
+			StateContext: func(
+				ctx context.Context,
+				d *schema.ResourceData,
+				i interface{},
+			) ([]*schema.ResourceData, error) {
+				id := d.Id()
+				err := d.Set("id", id)
+				if err != nil {
+					return nil, fmt.Errorf("failed to set 'id': %w", err)
+				}
+				return []*schema.ResourceData{d}, nil
+			},
+		},
+
+		Schema: map[string]*schema.Schema{
+			"id": {
+				Description: "Terraform ID of this resource",
+				Type:        schema.TypeString,
+				Computed:    true,
+			},
+
+			"name": {
```

Comment:
Lets create constants for the schema keys (see `resource_cyral_repository` for an example) and use them on d.Set and d.Get and other places that might be needed.
```suggestion
			AWSIAMIntegrationNameKey: {
```